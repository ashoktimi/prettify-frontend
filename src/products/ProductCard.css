
.ProductCard{
  position: relative;
  z-index: 1;
  height: 95%;
  width: 100%;
  background-color: white;
  display: flex;
  flex-direction: column;
}
.ProductCard-imgDiv{
  height: 300px;
  overflow: hidden;
  background-color: rgb(219, 204, 205);
}
.img{
  object-fit: cover;
  height: 100%;
  width: 100%;
  padding-top: 2%;
  background-color: rgb(255, 252, 252);
}

.ProductCard-body{
  height: 150px;
  border-radius: .5rem;
  overflow: hidden;
  background-color: white;
  width: 80%;
  left: 0;
  right: 0;
  margin: auto;
  position: relative;
}

.productAddBtn{
  color: black;
  position: absolute; 
  bottom: 12px;
  width: 95%;
  background-color: rgb(37, 150, 190);
}

.ProductCard-price {
font-size: 15px;
margin: 0;
color: black;
display: inline-block;  
font-weight: bold;
}
.priceBefore{
text-decoration: line-through;
}
.ProductCard-rating{
display: flex;
justify-content: space-between;
}
.ProductCard-star {
margin-right: 0.5rem; 
}
.Product-count{
display: inline-block;
}