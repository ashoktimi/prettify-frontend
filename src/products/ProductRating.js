import React from 'react';
import { faStar, faStarHalfAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';


const ProductRating = ({ rating }) =>{

  const fullStars = Math.floor(rating);
  const partialStar = rating - fullStars;
  const emptyStars = 5 - fullStars - Math.ceil(partialStar);
  
  return (
      <>
        {Array.from({ length: fullStars }, (_, i) => (
            <FontAwesomeIcon
              key={i}
              icon={faStar}
              className="ProductDetail-star"
              style={{ color: '#fdd835' }}
            />
          ))}
          {partialStar > 0 && (
            <FontAwesomeIcon
              icon={faStarHalfAlt}
              className="ProductDetail-star"
              style={{ color: '#fdd835' }}
            />
          )}
          {Array.from({ length: emptyStars }, (_, i) => (
            <FontAwesomeIcon
              key={i + fullStars}
              icon={faStar}
              className="ProductDetail-star"
              style={{ color: 'lightgray' }}
            />
          ))}
     </>
    )  
}
  
export default ProductRating;