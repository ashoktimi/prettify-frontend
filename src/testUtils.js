import React from "react";
import UserContext from "./auth/userContext";


const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, handleToggleOffcanvas = () => false }) => (
    <UserContext.Provider value={{ currentUser, handleToggleOffcanvas }}>
      {children}
    </UserContext.Provider>
);

function setup(){
  window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  });
}

function teardown(){
   window.matchMedia.mockClear();
   delete window.matchMedia;
}

function commonBeforeEach() {
  setup();
}

function commonAfterEach() {
  teardown();
}

export { UserProvider, commonBeforeEach, commonAfterEach };