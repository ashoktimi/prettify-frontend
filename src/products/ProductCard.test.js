import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { UserProvider,  commonBeforeEach, commonAfterEach } from "../testUtils";
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from '../cart/cartSlice';
import ProductCard from "./productCard";
import ProductRating from './ProductRating';

beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

describe('ProductCard', () => {
 const store = configureStore({
   reducer: {
     cart: cartReducer
   }
 });

 test("renders without crashing", function () {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <UserProvider>
            <ProductCard />
          </UserProvider>          
        </Provider>    
      </MemoryRouter>   
     );
  });
  
  test("matches snapshot", function () {
    const { asFragment } = 
     render(
      <MemoryRouter>
        <Provider store={store}>
         <UserProvider>
           <ProductCard />
         </UserProvider>          
        </Provider>    
      </MemoryRouter>  
     );
    expect(asFragment()).toMatchSnapshot();
  });
    
  test("ProductRating component renders correctly ", function () {
    render(
      <MemoryRouter>
         <ProductRating />
      </MemoryRouter>   
     );
  });
})
