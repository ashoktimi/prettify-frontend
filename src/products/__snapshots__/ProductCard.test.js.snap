// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProductCard card"
  >
    <div
      class="ProductCard-imgDiv"
    />
    <div
      class="ProductCard-body card-body"
    >
      <div
        class="card-title h5"
        style="text-align: center;"
      />
      <div
        style="display: flex; justify-content: center;"
      >
        <small
          class="ProductCard-price"
        >
           $ 
        </small>
      </div>
      <div
        class="ProductCard-rating"
      >
        <small>
          ()
        </small>
        <p>
           () Reviews 
        </p>
      </div>
      <button
        class="productAddBtn btn"
        type="button"
      >
        add to cart
      </button>
    </div>
  </div>
</DocumentFragment>
`;
